libtock/l3gd20.c

```diff
@@ -0,0 +1,140 @@
+#include "l3gd20.h"
+
+typedef struct l3gd20_response {
+  int data1;
+  int data2;
+  int data3;
+  bool done;
+} L3GD20Response;
+
+const float SENZIVITY_FACTOR[3] = {8.75, 17.5, 70.0};
+
+unsigned char senzitivity_factor = 0;
+
+static void command_callback_yield (int data1, int data2, int data3, void* ud) {
+  L3GD20Response *response = (L3GD20Response*)ud;
+  if (response) {
+    response->data1 = data1;
+    response->data2 = data2;
+    response->data3 = data3;
+    response->done  = true;
+  }
+}
+
+static int l3gd20_subscribe (subscribe_cb cb, void *userdata) {
+  return subscribe(DRIVER_NUM_L3GD20, 0, cb, userdata);
+}
+
+bool l3gd20_is_present (void) {
+  L3GD20Response response;
+  response.data1 = false;
+  response.done  = false;
+  // subscribe
+  l3gd20_subscribe (command_callback_yield, &response);
+  if (command (DRIVER_NUM_L3GD20, 1, 0, 0) == TOCK_SUCCESS) {
+    yield_for (&(response.done));
+  }
+  // unsubscribe
+  l3gd20_subscribe (NULL, NULL);
+  return response.data1 ? true : false;
+}
+
+int l3gd20_power_on (void) {
+  int evalue;
+  L3GD20Response response;
+  response.done = false;
+  // subscribe
+  l3gd20_subscribe (command_callback_yield, &response);
+  evalue = command (DRIVER_NUM_L3GD20, 2, 0, 0);
+  if (evalue == TOCK_SUCCESS) {
+    yield_for (&(response.done));
+  }
+  // unsubscribe
```

Comment:
I don't think you should need to unsubscribe here or in the other functions -- it doesn't really hurt, but it doesn't do anything meaningful either